--- smokeping-2.7.3/lib/Smokeping/Slave.pm	2011-10-05 17:51:46.000000000 +0800
+++ smokeping/lib/Smokeping/Slave.pm	2022-05-21 10:53:49.633959600 +0800
@@ -107,9 +107,16 @@
             warn "WARNING $slave_cfg->{master_url} sent data with wrong key";
             return undef;
         }
+        # Safe seems to reset SIG on at least FreeBSD, causing slave to crash after first reload
+        # since all handlers are gone.
+        my %sig_backup = %SIG;
+
         my $zone = new Safe;
         # $zone->permit_only(???); #input welcome as to good settings
         my $config = $zone->reval($data);
+
+        %SIG = %sig_backup;
+        
         if ($@){
             warn "WARNING evaluating new config from server failed: $@ --\n$data";
         } elsif (defined $config and ref $config eq 'HASH'){
